contrib/gin-gonic/gin/gintrace.go

```diff
@@ -16,7 +16,12 @@ import (
 func Middleware(service string) gin.HandlerFunc {
 	return func(c *gin.Context) {
 		resource := c.HandlerName()
+
+		// try to get a span context from request headers
+		sctx, _ := tracer.Extract(tracer.HTTPHeadersCarrier(c.Request.Header))
```

Comment:
Don't ignore the error, it is there for a reason. You can get inspired from here on how to accomplish this: https://github.com/DataDog/dd-trace-go/blob/v1/contrib/internal/httputil/trace.go#L25-L27